src/flame/variations/types.ts

```diff
@@ -0,0 +1,58 @@
+import tgpu, { TgpuFn } from 'typegpu'
+import { AnyWgslData, f32, Infer, struct, Vec2f, vec2f, vec4u } from 'typegpu/data'
+
+export type SimpleFunction = {
```

Comment:
Let's directly name these `SimpleVariation`, `DependentVariation` etc. and for helper fns, `parametricVar` or `parametricVariation`

Reply:
Refactored all